### PR TITLE
Marketplace Routing + Shop by farm

### PR DESCRIPTION
**Pull request by DiyaBaliga**
- marketplace routing w/ breadcrumbs
- dynamic routes for shopping by farm/ searching (accounts for filter/search issues0
- shop by farm tab w/ produce listings + updated labelling

issues:
- currently searching and filtering do not work on shop by farm tab
- searches have a bug where listings are not showing up, likely  due to async issues
- filtering sidebar is built around just the produce listings so it needs to be updated to account for tab switching (showing number fitting filters for each tab etc) as well as adapting current filter methods to also filter the shopByFarm listings (should be easy provided there are no async issues)

### CODE REVIEW

File: src/components/Marketplace/MarketplaceBreadcrumbs.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    } else if (searchTerms) {
+      const searchLink = `/marketplace/${searchTerms}`;
+      tempBread[searchLink] = { name: searchText, linkable: true };
+    }
+    return new Promise((resolve) => resolve(tempBread));
+  }
+
+  // map setup helper
+  async function setup() {
+    // pathnames = history.location.pathname.split('/').filter((x) => x);
```
Comment (hwang635) [inline src/components/Marketplace/MarketplaceBreadcrumbs.js]: Remove unused comments

File: src/components/Marketplace/MarketplaceBreadcrumbs.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { Link, useHistory } from 'react-router-dom';
```
Comment (hwang635) [inline src/components/Marketplace/MarketplaceBreadcrumbs.js:2]: This is a tough feature, great job implementing it!

File: src/components/Marketplace/MarketplaceScreen.js [hunk 3]
```diff
@@ -30,61 +33,166 @@ const INITIAL_POPUP_PRODUCE = {
...
+  const filterBySearch = () => {
+    filterProducebySearch(allProduceListings, setProduceListings);
+    if (shopByFarmID === undefined) {
+      filterFarmsBySearch();
+    }
+    // else the shopByFarm method is called and will filter after fetching records
+  };
+
+  // Get prod id, grouped unit type, price per grouped unit for each of a farm's produce records
+  // run if shopping by farm
```
Comment (hwang635) [inline src/components/Marketplace/MarketplaceScreen.js]: What does this comment mean?

File: src/components/Marketplace/MarketplaceScreen.js [hunk 4]
```diff
@@ -30,61 +33,166 @@ const INITIAL_POPUP_PRODUCE = {
...
+                });
+              } else {
+                tempProduce = tempProduce.concat(recordInfo);
+              }
+              setShopByFarmProduce(tempProduce);
+              setFilteredShopByFarmProduce(tempProduce);
+            });
+          });
+        } else {
+          console.log('no listings for this farm');
```
Comment (hwang635) [inline src/components/Marketplace/MarketplaceScreen.js]: Remove console logs

File: src/components/Marketplace/MarketplaceScreen.js [hunk 5]
```diff
@@ -1,11 +1,14 @@
+/* eslint-disable react/prop-types */
```
Comment (hwang635) [inline src/components/Marketplace/MarketplaceScreen.js:1]: Is this disable still necessary?

**Review by hwang635 — CHANGES_REQUESTED** [review]
Hey Diya, thanks so much for working on this feature! The breadcrumbs and search are pretty confusing and you implemented them really well. I didn't notice any issues besides the ones you're already aware of, so it looks good. I commented on a few minor things; if you could also merge the conflicts with master then your pr should be good to commit.

File: src/components/Marketplace/MarketplaceScreen.js (hunk 3 above)
Comment (DiyaBaliga) [reply src/components/Marketplace/MarketplaceScreen.js]: I meant that this method grabs all the produce record info for a given farmID and that it's called when shopping by farm, but the comment is pretty unclear so I'll edit it! if you have any preferences for how it should be phrased lmk!

**Review by hwang635 — CHANGES_REQUESTED** [review]
Note for the search/breadcrumbs/async bug we noticed.

**Review by hwang635 — APPROVED** [review]

**Pull request merged**